Lib/test/test_codecs.py

```diff
@@ -1641,6 +1641,12 @@ def test_register(self):
         self.assertRaises(TypeError, codecs.register)
         self.assertRaises(TypeError, codecs.register, 42)
 
+    def test_unregister(self):
+        def test_encoding(encoding):
+            return None
+        codecs.register(test_encoding)
+        self.assertEqual(codecs.unregister(test_encoding), None)
```

Comment:
Hum, you can use a mock.Mock object and ensure that looking for an unknown encoding name doesn't call the mock, after the function is unregistered?